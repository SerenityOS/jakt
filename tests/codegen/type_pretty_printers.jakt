/// Expect:
/// - output: "Value::Bar(values: [\"hello\"])\nRect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))\nFoo::Baz(100)\n"

class Size {
    public width: i64
    public height: i64
}

class Point {
    public x: i64
    public y: i64
}

class Rect {
    public location: Point
    public size: Size
}

enum Value {
    Foo
    Bar(values: [String])
}

enum Foo {
    Bar
    Baz(i64)
}

fn main() {
    let e = Value::Bar(values: ["hello"])
    println("{}", e)

    let r = Rect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))
    println("{}", r)

    let foo = Foo::Baz(100)
    println("{}", foo)
}
