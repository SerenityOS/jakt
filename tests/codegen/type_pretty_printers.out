Value::Bar(values: ["hello"])
Rect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))
Foo::Baz(100)
